### PR TITLE
storage: add tests

### PR DESCRIPTION
**Pull request by disksing**
@ngaut @siddontang @zhangjinpeng1987 @BusyJay

### CODE REVIEW

File: tests/storage/test_storage.rs [hunk 1]
```diff
@@ -183,6 +195,15 @@ fn test_txn_store_cleanup_commit() {
 }
 
 #[test]
+fn test_txn_store_batch_get() {
+    let store = new_assertion_storage();
+    store.put_ok(b"x", b"x1", 5, 10);
+    store.put_ok(b"y", b"y1", 15, 20);
+    store.put_ok(b"z", b"z1", 25, 30);
+    store.batch_get_ok(&[b"x", b"y", b"z"], 20, vec![b"x1", b"y1"]);
```
Comment (ngaut) [inline tests/storage/test_storage.rs]: How about add more assert on timestamp 14, 15, 16 and 19, 20, 21

File: src/storage/engine/mod.rs [hunk 2]
```diff
@@ -381,8 +393,22 @@ mod tests {
...
+        let mut cursor = snapshot.iter(None).unwrap();
+        let cursor_mut = cursor.as_mut();
+        assert_near_seek(cursor_mut, b"x", (b"x", b"1"));
+        assert_near_seek(cursor_mut, b"z", (b"z", b"2"));
+
         must_delete(engine, b"x");
         must_delete(engine, b"z");
+        for i in 0..100 {
+            let key = format!("y{}", i);
+            must_delete(engine, &key.as_bytes());
```
Comment (zhangjinpeng1987) [inline src/storage/engine/mod.rs]: What's the purpose?

Comment (disksing) [inline src/storage/engine/mod.rs]: Insert many key-values between 'x' and 'z' then`near_seek` will fallback to `seek`.

Comment (ngaut) [inline src/storage/engine/mod.rs]: Please add comments.
And why 100 ?

File: src/storage/engine/mod.rs [hunk 3]
```diff
@@ -369,8 +381,23 @@ mod tests {
         assert_near_seek(cursor_mut, b"y", (b"z", b"2"));
         assert_near_seek(cursor_mut, b"x\x00", (b"z", b"2"));
         assert!(!cursor_mut.near_seek(&make_key(b"z\x00")).unwrap());
+        // Insert many key-values between 'x' and 'z' then near_seek will fallback to seek.
```
Comment (siddontang) [inline src/storage/engine/mod.rs]: how can we ensure that near_seek will fall back to seek?

Comment (disksing) [inline src/storage/engine/mod.rs]: Only by checking test coverage we can know it.

File: src/storage/engine/mod.rs [hunk 4]
```diff
@@ -369,8 +381,23 @@ mod tests {
...
+        }
+        let snapshot = engine.snapshot(&Context::new()).unwrap();
+        let mut cursor = snapshot.iter(None).unwrap();
+        let cursor_mut = cursor.as_mut();
+        assert_near_seek(cursor_mut, b"x", (b"x", b"1"));
+        assert_near_seek(cursor_mut, b"z", (b"z", b"2"));
+
         must_delete(engine, b"x");
         must_delete(engine, b"z");
+        for i in 0..super::SEEK_BOUND{
```
Comment (zhangjinpeng1987) [inline src/storage/engine/mod.rs]: SEEK_BOUND+1?

File: src/storage/engine/mod.rs (hunk 3 above)
Comment (siddontang) [inline src/storage/engine/mod.rs]: Seem that we can use CURSOR_OVER_SEEK_BOUND_COUNTER to check it.

Comment (disksing) [inline src/storage/engine/mod.rs]: I think that would be very strange. In addition, tests could run in multiple threads, we will not able to confirm it is increased by this test.

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**